### PR TITLE
Add unmap method to dispatcher

### PR DESCRIPTION
**Pull request by DrLuke**
I noticed that there was no way to unmap handlers from addresses, so I added one. Comes with unit test.

### CODE REVIEW

File: pythonosc/dispatcher.py [hunk 1]
```diff
@@ -54,6 +54,19 @@ def map(self, address, handler, *args, needs_reply_address=False):
...
+
+    Args:
+      - address: An explicit endpoint.
+      - handler: A function that will be run when the address matches with
+                 the OscMessage passed as parameter.
+      - args: Any additional arguments that will be always passed to the
+              handlers after the osc messages arguments if any.
+      - needs_reply_address: True if the handler function needs the
+              originating client address passed (as the first argument)."""
+    self._map[address].remove(Handler(handler, list(args), needs_reply_address))
```
Comment (attwad) [inline pythonosc/dispatcher.py]: check for presence to avoid a KeyError exception being thrown
```
if not address in self._map:
  return
```

File: pythonosc/dispatcher.py [hunk 2]
```diff
@@ -54,6 +54,19 @@ def map(self, address, handler, *args, needs_reply_address=False):
     # regarding multiple mappings
     self._map[address].append(Handler(handler, list(args), needs_reply_address))
 
+  def unmap(self, address, handler, *args, needs_reply_address=False):
```
Comment (attwad) [inline pythonosc/dispatcher.py]: this signature feels wrong, there should be no need to reconstruct a handler just to remove it from the map, usually callers will have the handler instance already accessible and they can pass it directly to this method.
Otherwise another cleaner method to get it would be to call `handlers_for_address` and pass the returned handler(s) in the response.

basically: `def unmap(self, address, handler)`

File: pythonosc/test/test_dispatcher.py [hunk 3]
```diff
@@ -121,5 +121,14 @@ def test_multiple_handlers_with_wildcard_map(self):
     self.sortAndAssertSequenceEqual(
         [Handler(1, []), Handler(2, [])], self.dispatcher.handlers_for_address("/foo/bar"))
 
+  def test_unmap(self):
```
Comment (attwad) [inline pythonosc/test/test_dispatcher.py]: please add a test that checks for the key presence as I hinted before (you can use an empty dispatcher with no handlers mapped for example)

File: pythonosc/dispatcher.py (hunk 1 above)
Comment (DrLuke) [reply pythonosc/dispatcher.py]: I also thought about this when writing the method, but I think it's more verbose to have the code throw the exception. Either you keep track of your mapped handlers, or you can handle the exception. If it were to just return without any message, you would get no feedback that your unmapping failed/did nothing.

File: pythonosc/dispatcher.py (hunk 2 above)
Comment (DrLuke) [reply pythonosc/dispatcher.py]: Maybe it would make sense to have the map method return the handler? That way you can directly plug it back into unmap. Otherwise I'd suggest having an overloaded method to allow both signatures.
With handlers_for_address you'd have to comb through the handlers again. Imagine having mutliple handlers for the same address but just with different *args arguments. It would add a lot of boilerplate code. Although this admittedly should be a really rare case.

**Comment by DrLuke** [comment]
I have overloaded the unmap method to accept both the same arguments as map does or a handler object. Thanks to the python3 overloading decorator this is communicated perfectly by any good autocompletion:
image

Additionally I've made the map method return the Handler object it creates, to make later unmapping easier.

Unittest for both unmap-types is included.

**Comment by DrLuke** [comment]
I've added a more verbose exception for trying to unmap unmapped handlers:
```
Traceback (most recent call last):
  File "/home/drluke/prog/python-osc/pythonosc/dispatcher.py", line 91, in unmap
    self._map[address].remove(handler)
ValueError: list.remove(x): x not in list

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/drluke/prog/python-osc/dispatchertest.py", line 14, in <module>
    d.unmap("foo", handler)
  File "/home/drluke/prog/python-osc/pythonosc/dispatcher.py", line 96, in unmap
    raise ValueError("Address '%s' doesn't have handler '%s' mapped to it" % (address, handler)) from e
ValueError: Address 'foo' doesn't have handler '<pythonosc.dispatcher.Handler object at 0x7ff2756a8b38>' mapped to it
```

I think this is the best way to handle this error but I'm open for suggestions on how to do it otherwise.

**Comment by attwad** [comment]
Can you pelase remove 3.4 form the travis config in this PR so that it's green and https://github.com/attwad/python-osc/issues/86 is unblocked? after that I'll merge, thanks for the fixes
